### PR TITLE
Adding inputmode numeric to integer fields

### LINKED ISSUE #17745: Input types should match the column type
### Is your feature request related to a problem? Please describe.

As a person using a mobile or tablet, I want the correct keyboard to pop up when I search or edit a field in phpMyAdmin.  At the moment, if I'm searching for a field which is an `int` or `float` the normal software keyboard appears and I have to manually switch it to number mode.

### Describe the solution you'd like

```html
<td data-type="bigint(20)">
   <input inputmode="numeric" type="text" id="fieldID_9" name="criteriaValues[9]" class="textfield">
   ...
```

### Describe alternatives you've considered

HTML 5 provides a couple of ways to do this, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number

### Additional context

- phpMyAdmin version: 4.9.7

If this is something you're interested in, I'm happy to provide a PR.

### PR DESCRIPTION
**Pull request by greew**
### Description

To easier use phpMyAdmin on touch devices with virtual keyboards, adding the inputmode ensures that the correct virtual keyboard is shown per default when focusing the field.

See https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode for more information.

Ref:  #17745 

Signed-off-by: Jesper Skytte <jesper@skytte.it>

### CODE REVIEW

**Review by williamdes — APPROVED** [review]

**Review by ibennetch — APPROVED** [review]

**Comment by greew** [comment]
As far as I can see, the failing test can't be on my part, as I haven't changed anything about the `$column` variable. 

Can you confirm that?

```
1) PhpMyAdmin\Tests\InsertEditTest::testGetHTMLinput
Undefined array key "Type"

/app/libraries/classes/InsertEdit.php:699
/app/test/classes/AbstractTestCase.php:223
/app/test/classes/InsertEditTest.php:1064
/app/vendor/phpunit/phpunit/phpunit:98
/app/vendor/bin/phpunit:123
```

File: test/classes/InsertEditTest.php [hunk 1]
```diff
@@ -1052,6 +1052,32 @@ public function testGetHTMLinput(): void
             . ' class="textfield datetimefield" c tabindex="25" id="field_0_3">',
             $result
         );
+
+        // case 4 int
+        $column['pma_type'] = 'int';
+        $column['True_Type'] = 'int';
```
Comment (MauricioFauth) [inline test/classes/InsertEditTest.php:1058]: The missing `$column['Type']` should be added.

```suggestion
        $column['True_Type'] = 'int';
        $column['Type'] = 'int(11)';
```

File: test/classes/InsertEditTest.php [hunk 2]
```diff
@@ -1052,6 +1052,32 @@ public function testGetHTMLinput(): void
...
+        $result = $this->callFunction(
+            $this->insertEdit,
+            InsertEdit::class,
+            'getHtmlInput',
+            [
+                $column,
+                'a',
+                'b',
+                30,
```
Comment (MauricioFauth) [inline test/classes/InsertEditTest.php:1067]:
```suggestion
                11,
```

File: test/classes/InsertEditTest.php [hunk 3]
```diff
@@ -1052,6 +1052,32 @@ public function testGetHTMLinput(): void
...
+                23,
+                2,
+                0,
+                'DATE',
+                false,
+            ]
+        );
+        $this->assertEquals(
+            '<input type="text" name="fieldsa" value="b" size="30" data-type="DATE"'
+            . ' class="textfield datetimefield" c tabindex="25" inputmode="numeric" id="field_0_3">',
```
Comment (MauricioFauth) [inline test/classes/InsertEditTest.php:1078]:
```suggestion
            '<input type="text" name="fieldsa" value="b" size="11" min="-2147483648" max="2147483647" data-type="INT"'
            . ' class="textfield" c tabindex="25" inputmode="numeric" id="field_0_3">',
```

File: libraries/classes/InsertEdit.php [hunk 4]
```diff
@@ -693,6 +693,7 @@ private function getHtmlInput(
             }
         }
 
+        $inputMode = '';
```
Comment (MauricioFauth) [inline libraries/classes/InsertEdit.php:696]:
```suggestion
```

File: libraries/classes/InsertEdit.php [hunk 5]
```diff
@@ -693,6 +693,7 @@ private function getHtmlInput(
...
         $inputMinMax = '';
         if (in_array($column['True_Type'], $this->dbi->types->getIntegerTypes())) {
```
Comment (MauricioFauth) [inline libraries/classes/InsertEdit.php:698]: You can assign the condition value to use it later to add the input mode. 

```suggestion
        $isInteger = in_array($column['True_Type'], $this->dbi->types->getIntegerTypes());
        if ($isInteger) {
```

File: libraries/classes/InsertEdit.php [hunk 6]
```diff
@@ -701,6 +702,7 @@ private function getHtmlInput(
             $inputMinMax = 'min="' . $minMaxValues[0] . '" '
                 . 'max="' . $minMaxValues[1] . '"';
             $dataType = 'INT';
+            $inputMode = 'inputmode="numeric"';
```
Comment (MauricioFauth) [inline libraries/classes/InsertEdit.php:705]:
```suggestion
```

File: libraries/classes/InsertEdit.php [hunk 7]
```diff
@@ -716,6 +718,7 @@ private function getHtmlInput(
             . ' data-type="' . $dataType . '"'
             . ' class="' . $theClass . '" ' . $onChangeClause
             . ' tabindex="' . ($tabindex + $tabindexForValue) . '"'
+            . ($inputMode ? ' ' . $inputMode : '')
```
Comment (MauricioFauth) [inline libraries/classes/InsertEdit.php:721]: That way the HTML is not stored in variable, so it's more clear what's happening here.

```suggestion
            . ($isInteger ? ' inputmode="numeric"' : '')
```

File: libraries/classes/InsertEdit.php [hunk 8]
```diff
@@ -693,6 +693,7 @@ private function getHtmlInput(
             }
         }
 
+        $inputMode = '';
         $inputMinMax = '';
         if (in_array($column['True_Type'], $this->dbi->types->getIntegerTypes())) {
             $extractedColumnspec = Util::extractColumnSpec($column['Type']);
```
Comment (MauricioFauth) [inline libraries/classes/InsertEdit.php:699]: The `$column['Type']` in this line is what caused the test to fail.

File: test/classes/InsertEditTest.php [hunk 9]
```diff
@@ -1052,6 +1052,32 @@ public function testGetHTMLinput(): void
...
+            [
+                $column,
+                'a',
+                'b',
+                30,
+                'c',
+                23,
+                2,
+                0,
+                'DATE',
```
Comment (MauricioFauth) [inline test/classes/InsertEditTest.php:1072]:
```suggestion
                'NUMBER',
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
Hi, @greew. Thank you for your contribution.

**Comment by greew** [comment]
> [MauricioFauth](/MauricioFauth)

Great stuff! I'll look into it right away! :)

File: libraries/classes/InsertEdit.php (hunk 8 above)
Comment (greew) [reply libraries/classes/InsertEdit.php:699]: That makes sense - thank you!

**Review by williamdes — APPROVED** [review]

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**